### DISCUSSION TITLE
Meeting: Celo Wallet and (d)App Council #15 [September] 💬

### DISCUSSION THREAD

**Discussion by arthurgousset** [celo-org/celo-monorepo#9774] [Wallet and (d)App Council]
# Monthly Celo Wallet and (d)App Council

##  📆 Quick links 

- When: **Wednesday, September 14 (9am PDT/6pm CEST)**
- Where: **[Google Meet](https://meet.google.com/uwj-wfse-uzm)**
- Agenda: **comments**👇
- Previous: https://github.com/celo-org/celo-monorepo/discussions/9695
- Next: https://github.com/celo-org/celo-monorepo/discussions/9875
- Recording: [Google Drive](https://drive.google.com/file/d/186a4KJv7NeHTFm4joKjYKshzzPjO1M7_/view?usp=sharing)

## 🤷‍♂️ What is this?

A monthly meeting for Celo **developers, founders,** **product** people (anyone really!) to come together and
- find consensus (coordinate),
- learn about needs, pain points, and desires
- present and collect technical feedback

## 🗒️  Agenda

- Comment with a short sentence describing what you'd like to discuss
- Don't worry, it can be small or big! 
- Every builder is welcome (and we are very easy going, I promise 😄)
- Upvote topics you're interested in

## Housekeeping

_You can safely ignore this!_

- [x] Send update calendar invite with Github link
- [x] Encourage additions to the agenda
- [x] Publicise on forum and Discord
- [x] RECORD meeting 🎥
- [x] Share recording below and on forum
- [x] Create next meeting (link to next meeting above)

**Comment by arthurgousset** [comment]
[ASv2 User Flows](https://www.notion.so/clabsco/ASv2-User-Flows-f508062cc99d4376af4e161859f2e745)

What is this? 
-  sequence diagrams describing typical interactions wallet users might make on client app or you might make on your backend server when interacting with Attestation v2

Ask: 
- [x] We'd love some feedback on these

**Reply by arthurgousset** [reply]
Eric - We reviewed it the day you sent it. 
Eric - Main thing we discussed was concern around 1 attester for **escrow and money transmitter**.
Eric - Our opinion is Argent's single guardian recovery solution should really require digital custody license.
Eric - Ideally we wouldn't run an attestation service atm

**Reply by arthurgousset** [reply]
Jean - I spent a couple hours on the flow atm
Jean - Question: **Will you provide an example implementation for the flows**?
Jean - Currently doing centralised phone number verification solution, shouldn't be hard to add ASv2
Jean - Design on paper is good, but in practice it sometimes turns out harder.
Jean - Probably 1mo until centralised phone number verification release (mid October)

**Comment by arthurgousset** [comment]
Crowdsourced agenda during meeting:
- [x] Subha: Ask to complete the cLabs developer feedback form: https://forms.gle/CHMkGo8B7t29vTTA9
- [x] Subha: DevX process
- [x] Patrick/Eric: Speeding up use-contractkit (react-celo)
  - [x] Eric - Has team considered forking rainbowkit?
- [x] Eric: Discuss progress on Minima (order routing)
- [ ] Jean: Curious about stCelo integration/changes
- [x] Patrick: Any risk with Ethereum merge discussion (e.g. bridges)

**Comment by arthurgousset** [comment]
Subha - Developer Experience topic

- We want to make sure you have all the tools you need (it's one of our top priorities)
- We'd love some feedback through this form: https://forms.gle/CHMkGo8B7t29vTTA9
- One of the Application team's OKRs is developer tools
- We have a form for developer requests, but we'd like to have a process around getting these requests with DevRel and Partner Eng
- We've already started getting feedback about developer tooling **documentation**

**Reply by arthurgousset** [reply]
Eric - Do you plan on sharing the raw data from the survey
Subha - We'll share the results if there are no privacy implications (e.g. remove emails)
Eric - Would be interested in taking a look

**Reply by arthurgousset** [reply]
Jean - Slightly confused about what happens between DevRel (Foundation) and cLabs
Subha - DevRel (Foundation) will take input from request form, DevRel will work with PartnerEng (cLabs). Currently we need to flesh out the details of the plan

**Reply by arthurgousset** [reply]
Jean - Feedback: Is there a plan on the **release schedule** for SDKs and new stuff.
Subha - We want to minimise breaking changes that require changes on dev side. We've been putting our milestones on the forum, so you can expect that transparency going forward. We've discussed the velocity in our team meeting.

**Comment by arthurgousset** [comment]
Eric: Discuss progress on Minima (order routing)

- Wanted to talk about the router they are working about on Node
- There were two routers
  - swappa
  - 0x (pretty bad implementation, e.g. doesn't support Moola swaps)
- Minima is like swappa but people can set custom fees (with API keys)
- Adv:
  - support more routes (get into LP positions, move in/out of stCELO via ideal route)
  - super fast (50% faster than swappa and 0x), we could save Ubeswap users 2h a day
  - path splitting (for large trades)
- Integrated:
  - Mobius
  - On Ubeswap
  - Submitted PR to Valora in-app swapping (currently uses 0x)
- Reach out to Eric via email for API 
- Check out docs: 
  - https://docs.router.nodefinance.org/

**Reply by arthurgousset** [reply]
Jean - Sounds great! Much better than 0x. We've implemented this in a cloud function, so should be able to switch quickly.
Eric - Met PM on Valora swaps in-person
Eric - Our PM is "do not merge" because it's a cloud function

**Reply by arthurgousset** [reply]
Patrick - Does this include limit orders
Eric - No, been chatting with Ubeswap about this

**Reply by arthurgousset** [reply]
Charlie - Just noticed 1-click yield farming, curious how it works
Charlie - Is the code open source? Been doing similar work on Revvo
Eric - Offchain route finding is not open source, but contracts are audited and labelled on Celo explorer

**Reply by arthurgousset** [reply]
Arthur - offchain vs onchain query
Eric - onchain does the whole path finding
Eric - offchain faster, easier integration
Charlie - seems like on chain would require oracles right?
Eric - Domain MEV person on Celo is also MEV person on Arbitrum
Charlie - We found that to avoid sandwich attack you need oracle prices

**Comment by arthurgousset** [comment]
Patrick: Any risk with Ethereum merge discussion (e.g. bridges)
- Unfortunately nobody from Optics team here
- Might potentially affect Ubeswap/Mobius

**Reply by arthurgousset** [reply]
Evan - Basically in our view bridged assets are in users' court
Evan - Haven't seen any messages from Optics on Eth merge (though biggest Ethereum TVL pool on Celo)

**Reply by arthurgousset** [reply]
Eric - Assuming Optics team is aware of this

**Reply by arthurgousset** [reply]
Evan - Probably good if Optics could put out a message about Eth merge risk

**Comment by arthurgousset** [comment]
Eric: People's thought on Mento spinout
- Seems like Celo is become less and less of a DeFi chain, and more of a ReFi chain
- Do we all plan to support mentoDollar when the spinout takes place

**Reply by arthurgousset** [reply]
Patrick
- P0 is making trading in/out of stable tokens easy
- I'd focus my resources on tokens that satisfy those

**Reply by arthurgousset** [reply]
Evan
- Mento seems to inherit regulatory risk, smart move from Foundation
- Curious what it means for Celo Reserve
- Celo put option was reserve
- I think it opens the doors to other tokens
- I agree that CICO should be the 1 priority

**Reply by arthurgousset** [reply]
Eric
- I'll start campaigning against this now
- I think it's a risk for the Celo community
- Risk is that Celo Reserve is owned by a more centralised entity (so more vulnerable to regulation)
- Makes little sense to give the reserve to Mento
- They will probably do a big airdrop

**Reply by arthurgousset** [reply]
Arthur - Don't understand the airdrop
Eric - There will be a massive airdrop to staking Celo
Eric - Still very interested in using Celo as financial applications layer

**Reply by arthurgousset** [reply]
Patrick - Concern is around fair distribution of mento token that will govern the Celo Reserve

**Reply by arthurgousset** [reply]
Eric - Query why Celo L1 should bolster Celo Reserve at protocol layer, if CELO token doesn't govern the reserve

**Comment by arthurgousset** [comment]
📽 Here is the [recording](https://drive.google.com/file/d/186a4KJv7NeHTFm4joKjYKshzzPjO1M7_/view?usp=sharing) for anyone that missed the meeting.

**Reply by Deepcryptodive** [reply]
Thanks for the thorough notes and recording. Much appreciated!

**Reply by arthurgousset** [reply]
You're most welcome! Feel free to join us at the next meeting 🙂
- https://github.com/celo-org/celo-monorepo/discussions/9875
